src/helpers.js

```diff
@@ -87,17 +87,34 @@ export function applySugar(environment: Object) {
 }
 
 export function getCommand(): { command: string, options: Object, parameters: Array<string> } {
+  // Print the environment separated by \0 in a POSIX compatible (!) way.
+  // Only environment variable names that consist only of alphanumeric
+  // characters and underscores, and begin with an alphabetic character or an
+  // underscore are included.
+  const shScript = ('env|' +
```

Comment:
> Correct me if I'm wrong but isn't it outputting the env in a \n seperated way and then we're using regexp to split it?

It outputs the env with ``\n`` and finds all names of potential environment variables with a regex. E.g. the environment
  * TEST
    ```
    text
    ```
  * TEST2
    ```
    text
    TEST3=text
    ```
would turn to ``TEST1``, ``TEST2`` and ``TEST3``. It then checks if this variables are defined in the shell (weeding out ``TEST3``). An indirect variable referenced is used to retrieve the value of the environment variables.

> Going to experiment with python env extraction because it's available by default on both Linux and macOS

I tried not to do any presumption about the environment it is running in and be as compatible as possible. If you are willing to presume the existence of Python this gets much easier of course.